### PR TITLE
add error enum & test reconstruction w/ insufficient shares

### PR DESCRIPTION
**Pull request by jhdewitt**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -58,9 +65,10 @@ fn share_value(t: u8, n: u8, secret_byte: &u8) -> Shares {
 // Vec<Share>, where each vec of shares belings to participant 1 -> n. t shares
 // are required to reconstruct the secret. `secret` is an arbitrary size byte
 // slice.
-pub fn construct_shares(t: u8, n: u8, secret: &[u8]) -> Result<Vec<Shares>, &str> {
+pub fn construct_shares(t: u8, n: u8, secret: &[u8]) -> Result<Vec<Shares>, SecretSharingError> {
     if t == 0 || n == 0 {
-        return Err("t and n can not be zero");
+        // return Err("t and n can not be zero");
```
Comment (avahowell) [inline src/lib.rs]: pls delet

File: src/lib.rs [hunk 2]
```diff
@@ -108,14 +116,17 @@ fn reconstruct_value(shares: Vec<&SharePoint>) -> gf::GF256e {
 // the shared secret. The reconstruction is not verifiable; reconstructing
 // invalid shares will return an invalid secret, not an error. Returns an error
 // if len(shares) < t
-pub fn reconstruct(t: u8, shares: Vec<Shares>) -> Result<Vec<u8>, &'static str> {
+// pub fn reconstruct(t: u8, shares: Vec<Shares>) -> Result<Vec<u8>, &'static str> {
+pub fn reconstruct(t: u8, shares: Vec<Shares>) -> Result<Vec<u8>, SecretSharingError> {
     if shares.len() < t as usize {
-        return Err("not enough shares to reconstruct secret");
+        // return Err("not enough shares to reconstruct secret");
```
Comment (avahowell) [inline src/lib.rs]: delet

File: src/lib.rs [hunk 3]
```diff
@@ -108,14 +116,17 @@ fn reconstruct_value(shares: Vec<&SharePoint>) -> gf::GF256e {
...
+        return Err(SecretSharingError::InsufficientShares);
     }
     let sz = shares[0].len();
     for i in 1..shares.len() {
         if shares[i].len() != sz {
-            return Err("missing share for byte");
+            // return Err("missing share for byte");
```
Comment (avahowell) [inline src/lib.rs]: delet

**Review by avahowell — CHANGES_REQUESTED** [review]

**Review by avahowell — APPROVED** [review]

**Comment by jhdewitt** [comment]
Thanks! ✅

**Pull request merged**